core/authn.go

```diff
@@ -34,26 +36,54 @@ type tokenResult struct {
 
 func (a *apiAuthn) handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		err := a.auth(req)
-		if err != nil {
-			errorFormatter.Write(req.Context(), rw, err)
+		token, err0 := a.tokenAuth(req)
+		ctx := req.Context()
+		if err0 == nil && token != "" {
+			// if this request was successfully authenticated, pass the token along
+			ctx = context.WithValue(ctx, "token", token)
+		}
+
+		err1 := a.localhostAuth(req)
```

Comment:
Instead of duplicating some of the logic from cmd/cored here, I think it would be better to delete it from cmd/cored entirely and rely just on this one.